src/Adapter/Product/Grid/Data/Factory/ProductGridDataFactoryDecorator.php

```diff
@@ -125,16 +151,21 @@ public function __construct(
         $this->taxEnabled = $taxEnabled;
         $this->isEcotaxEnabled = $isEcotaxEnabled;
         $this->ecoTaxGroupId = $ecoTaxGroupId;
+        $this->shopRepository = $shopRepository;
+        $this->productMultiShopRepository = $productMultiShopRepository;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getData(SearchCriteriaInterface $searchCriteria): GridData
     {
-        $productData = $this->productGridDataFactory->getData($searchCriteria);
+        if (!$searchCriteria instanceof ShopSearchCriteriaInterface) {
```

Comment:
Why not changing the parameter type instead?

Reply:
I can't because of the interface it implements, if I change the type it's no longer compatible..